services/graphql-server/utils/ast2sdl.go

```diff
@@ -81,3 +128,50 @@ func buildCustomArgumentsClause(builder *strings.Builder, rp graphql.ResolvePara
 	builder.WriteString(values)
 	builder.WriteString(")")
 }
+
+func buildFragmentsClause(query *sdlQuery, rp graphql.ResolveParams) {
+	for len(query.fragmentsToAdd) > 0 {
+		fragmentName := query.fragmentsToAdd[0]
+		query.addedFragments[fragmentName] = true
+		query.fragmentsToAdd = query.fragmentsToAdd[1:]
+		fragment := rp.Info.Fragments[fragmentName].(*ast.FragmentDefinition)
+
+		query.builder.WriteString(fmt.Sprintf("fragment %s on %s", fragmentName, fragment.TypeCondition.Name.Value))
+		resolveParamsToQueryInner(query, fragment, rp)
+	}
+}
+
+func buildVariableClause(query *sdlQuery) string {
```

Comment:
Everything here is called `buildXYZ` and it straight up adds stuff to the string builder
This one returns a string instead? And I'm not really sure what it does

Reply:
Variable clause is located in the beginning of the query but can be build only after all query is built so I used another builder for it and then it the whole query string is built from variable clause and fields and other parts

Reply:
Sure that's fine, but all the other `buildXYZ` methods add stuff to the string builder on the `sdlQuery` struct, whereas this one returns a string.
If they're all called `buildXYZ` I'd expect them to behave the same way, and they don't.

Reply:
I can rename it to `getVariableClause`. Does it be ok for you?

Reply:
Talked offline, decided on renaming the other functions to `writeXYZ` instead of `buildXYZ`